File: python/Easy/1512-number-of-good-pairs.py
class Solution:
    def numIdenticalPairs(self, nums: List[int]) -> int:
        seen = {}
        good_pairs =0 
        for num in nums:
            if num in seen: 
                good_pairs += seen[num]
                seen[num] += 1
            else:
                seen[num] = 1
        return good_pairs
        

File: javascript/Easy/1512-number-of-good-pairs.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var numIdenticalPairs = function(nums) {
    const seen = {}
    let goodPairs = 0
    for (let index=0; index<nums.length; index++) {
        if (nums[index] in seen) {
            goodPairs += seen[nums[index]]
            seen[nums[index]] += 1
        }
        else {
            seen[nums[index]] = 1
        }
    }
    return goodPairs
};